File: UnmanagedToManaged/Engine/Controls/CodeBox.vb
' Copyright (c) Microsoft Corporation.  All rights reserved.
Imports System.Drawing
Imports System.Windows.Forms

Namespace Controls

    Public Class CodeBox

        Public ReadOnly Property RichTextBox() As RichTextBox
            Get
                Return m_box
            End Get
        End Property

        Public Property Code() As String
            Get
                Return m_box.Text
            End Get
            Set(ByVal value As String)
                m_box.Text = value
                m_box.ScrollToCaret()
            End Set
        End Property

        Sub New()

            ' This call is required by the Windows Form Designer.
            InitializeComponent()

            ' Add any initialization after the InitializeComponent() call.
            Dim color As Color = m_box.BackColor
            m_box.ReadOnly = True
            m_box.BackColor = color
        End Sub

        Public Sub SelectAll()
            m_box.SelectAll()
        End Sub

        Public Sub Copy()
            m_box.Copy()
        End Sub

        Private Sub OnCopyClick(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles CopyToolStripMenuItem.Click
            m_box.Copy()
        End Sub

        Private Sub OnSelectAllClick(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles SelectAllToolStripMenuItem.Click
            m_box.SelectAll()
        End Sub
    End Class

End Namespace

File: PInvoke.Core/Controls/CodeBox.cs
// Copyright (c) Microsoft Corporation.  All rights reserved.

using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Drawing;
using System.Windows.Forms;

namespace PInvoke.Controls
{
    public partial class CodeBox
    {
        public RichTextBox RichTextBox
        {
            get { return m_box; }
        }

        public string Code
        {
            get { return m_box.Text; }
            set
            {
                m_box.Text = value;
                m_box.ScrollToCaret();
            }
        }


        public CodeBox()
        {
            // This call is required by the Windows Form Designer.
            InitializeComponent();

            // Add any initialization after the InitializeComponent() call.
            Color color = m_box.BackColor;
            m_box.ReadOnly = true;
            m_box.BackColor = color;
        }

        public void SelectAll()
        {
            m_box.SelectAll();
        }

        public void Copy()
        {
            m_box.Copy();
        }

        private void OnCopyClick(System.Object sender, System.EventArgs e)
        {
            m_box.Copy();
        }

        private void OnSelectAllClick(System.Object sender, System.EventArgs e)
        {
            m_box.SelectAll();
        }
    }
}
